### PR TITLE
My First Half-Stack App: In Progress

### PR DESCRIPTION
**Pull request by VerteDinde**
Going for bonus points next.

### CODE REVIEW

File: lib/app.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  '404': notFound
+};
+
+function app(req, res) {
+  const url = parsePath(req.url) || notFound;
+  req.query = url.query;
+  req.params = url.params;
+
+  res.setHeader('Content-Type', 'application/JSON');
+  const route = routes[url.route];
```
Comment (dchasepdx) [inline lib/app.js]: There's an opportunity to handle 404 errors here. What happens if someone enters a resource that doesn't exist?

File: lib/routes/puppies.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          }
+          const serialPom = JSON.stringify(pom);
+          res.end(serialPom);
+        });
+    }
+
+    poms.find()
+      .toArray()
+      .then(poms => {
+        if (!poms) {
```
Comment (dchasepdx) [inline lib/routes/puppies.js]: If there's nothing in the poms collections, it's not a 404 resource not found, you should return an empty object. 404 should be used for resources that don't exist in your api

File: lib/app.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const parsePath = require('./helpers/parsePath');
+const notFound = require('./helpers/notFound');
+const puppies = require('./routes/puppies');
+
+const routes = {
+  'puppies': puppies,
+  '404': notFound
```
Comment (dchasepdx) [inline lib/app.js]: Not clear on why you have a 404 route. You could potentially redirect bad requests to this 404 path, but we're not there yet.

File: lib/routes/puppies.js (hunk 2 above)
Comment (VerteDinde) [inline lib/routes/puppies.js]: Oh interesting, okay - let me try that

File: lib/app.js (hunk 3 above)
Comment (VerteDinde) [inline lib/app.js]: Sorry, this was just for testing - removed!

**Pull request merged**